commentlock.py

```diff
@@ -1,92 +1,124 @@
 import requests
-import re
-import time
-import json
-import math
+from time import sleep
+from getpass import getpass
 
-# Get user info
-print("Enter your username:")
-username = input()
+def prepare_session(username: str, password: str) -> requests.Session:
+    # Reference: https://github.com/yuwex/scratchcloud/blob/main/scratchcloud/client.py#L362
 
-print("Enter your password:")
-password = input()
-
-print("Should comments be enabled? Y/n")
-comments_on_yn = input()
+    # Cookie header isn't needed
+    headers = {
+        "X-CSRFToken": "None",
+        "X-Requested-With": "XMLHttpRequest",
+        "Referer": "https://scratch.mit.edu",
+        "User-Agent": "commentlock",
+    }
 
-if comments_on_yn == "y":
-    comments_on = True
-else:
-    comments_on = False
+    # Create session
+    session = requests.Session()
+    session.headers = headers
 
-with requests.Session() as s:
+    # Get CSRF first instead of making one up. This is a best practice.
+    resp = session.get("https://scratch.mit.edu/csrf_token/", headers=headers)
 
-    # https://github.com/CubeyTheCube/scratchclient/tree/main/scratchclient
-    headers = {
-        "x-csrftoken": "a",
-        "x-requested-with": "XMLHttpRequest",
-        "Cookie": "scratchcsrftoken=a;scratchlanguage=en;",
-        "referer": "https://scratch.mit.edu",
-        "user-agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.101 Safari/537.36"
-    }
-    data = json.dumps({"username": username,"password": password,"useMessages": "true"})
+    # Use built in resp.cookies.get() instead of regex.
+    csrf_token = resp.cookies.get("scratchcsrftoken")
 
-    # Login with user info
-    r = s.post("https://scratch.mit.edu/login/", data=data, headers=headers)
-    print(r.status_code)
-    session_id = re.search('"(.*)"', r.headers["Set-Cookie"]).group()
-    token = r.json()[0]["token"]
+    # Update headers to include real CSRF
+    session.headers["X-CSRFToken"] = csrf_token
 
-    # Set headers
-    headers = {
-            "x-requested-with": "XMLHttpRequest",
-            "Cookie": "scratchlanguage=en;permissions=%7B%7D;",
-            "referer": "https://scratch.mit.edu"
+    # Prepare login payload
+    payload = {
+        "username": username,
+        "password": password,
     }
 
-    # Get CSRF token
-    r = s.get("https://scratch.mit.edu/csrf_token/", headers=headers)
-    print(r.status_code)
-    csrf_token = re.search(
-            "scratchcsrftoken=(.*?);", r.headers["Set-Cookie"]
-    ).group(1)
+    # Log in with CSRF Token.
+    resp = session.post(
+        "https://scratch.mit.edu/login/",
+        json=payload,

```

Comment:
Should the equals sign have spaces around it?

Reply:
The spaces are purely stylistic. I ran this code through [black](https://black.readthedocs.io/en/stable/), and it decided to remove the spaces as well as all of the other formatting.